### PR TITLE
`consistent-function-scoping`: Ignore within `useEffect` in React Hooks

### LINKED ISSUE #392: `consistent-function-scoping`: Ignore within `useEffect` in React Hooks
In React Hooks, I got:

```js
useEffect(() => {
  async function getItems() {}
  
  getItems()
}, [])
```

And it's definitely better to keep the function in the `useEffect`, to group it together. But this rule recommends moving out the function, of course.

Perhaps perform a check if the function is in `useEffect`, to ignore it?

### PR DESCRIPTION
**Pull request by fisker**
Fixes #392

### CODE REVIEW

File: test/consistent-function-scoping.js [hunk 1]
```diff
@@ -362,6 +368,25 @@ ruleTester.run('consistent-function-scoping', rule, {
...
+				}, [])
+			`,
+			errors: [
+				{
+					ruleId: 'consistent-function-scoping',
+					messageId: 'FunctionDeclaration',
+					column: 3,
+					line: 3
+				}
+			]
```
Comment (fisker) [inline test/consistent-function-scoping.js]: Todo: use function name to test after merge https://github.com/sindresorhus/eslint-plugin-unicorn/pull/589

File: rules/consistent-function-scoping.js [hunk 2]
```diff
@@ -60,6 +60,13 @@ function checkReferences(scope, parent, scopeManager) {
 		);
 }
 
+const isReactHooks = scope =>
```
Comment (sindresorhus) [inline rules/consistent-function-scoping.js]:
```suggestion
const isReactHook = scope =>
```

It's only a single hook.

File: rules/consistent-function-scoping.js [hunk 3]
```diff
@@ -60,6 +60,13 @@ function checkReferences(scope, parent, scopeManager) {
...
+	scope.block &&
+	scope.block.parent &&
+	scope.block.parent.callee &&
+	scope.block.parent.callee.type === 'Identifier' &&
+	scope.block.parent.callee.name === 'useEffect';
```
Comment (sindresorhus) [inline rules/consistent-function-scoping.js]: There are more built-in hooks: https://reactjs.org/docs/hooks-reference.html

Comment (fisker) [reply rules/consistent-function-scoping.js]: Added https://github.com/sindresorhus/eslint-plugin-unicorn/pull/588/commits/4a4f06efd7e67ef779017cbf4b0c8cc8bd3cae66

Feel I don't need remove those hooks doesn't accept function parameter.

**Pull request merged**